bluebottle/surveys/adapters.py

```diff
@@ -139,18 +139,21 @@ def update_survey(self, survey):
                 }
             )
             params = self.parse_query_params(response)
-            if params.has_key('project_id') and params['project_id']:
+            # Store all params
+            resp.params = params
+
+            # Find and store project/task
+            if 'project_id' in params and params['project_id']:
                 try:
                     resp.project = Project.objects.get(pk=int(params['project_id']))
                 except Project.DoesNotExist:
                     pass
-            if params.has_key('task_id') and params['task_id']:
+            if 'task_id' in params and params['task_id']:
```

Comment:
You could do:

try:
    resp.task = Task.objects.get(pk=int(params['task_id']))
except KeyError, Task.DoesNotExist:
    pass